### ISSUE TITLE
Support applying parquet bloom filters to StringView columns

### ISSUE THREAD

**Issue by alamb** [apache/datafusion#12499]
### Is your feature request related to a problem or challenge?

Part of https://github.com/apache/datafusion/issues/11752

While working to enable StringView in https://github.com/apache/datafusion/pull/12092 I found that the columns when read as `StringView` and `BinaryView` do not take advantage of Bloom filters. 

Specifically this code doesn't handle StringView
https://github.com/apache/datafusion/blob/a08f923c2acb1a46614970231d9a672c36ce3ad2/datafusion/core/src/datasource/physical_plan/parquet/row_group_filter.rs#L267-L272

### Describe the solution you'd like

Support applying parquet bloom filters to StringView columns

### Describe alternatives you've considered

Basically:
1. Make the code changes for bloom filters in https://github.com/apache/datafusion/pull/12092
2. Write a test 

In terms of testing, I think the easiest thing to do would be to follow the model of the existing tests for Utf8/Binary columns and pass the `schema_force_view_types ` config flag

**Comment by alamb** [comment]
I think this is a clearly defined need so marking as good first issue

**Comment by my-vegetable-has-exploded** [comment]
take
